### DISCUSSION TITLE
Flow Deck Not Working

### DISCUSSION THREAD

**Discussion by meesher1** [bitcraze/discussions#502] [Q&A]
When I attach the flow deck to my crazyflie I got an error in the console stating that my deck has corrupt memory and it cannot be initialized. 

```
SYS: Build 0:b8d2e0b9e983 (2022.12) CLEAN
SYS: I am 0x3530343431315103002D0030 and I have 1024KB of flash
CFGBLK: v1, verification [OK]
DECK_INFO: Memory error: wrong header ID
DECK_INFO: Deck 0 has corrupt OW memory. No driver will be initialized!
DECK_CORE: 0 deck(s) found
```

Does anyone have recommendations to resolve this?

**Comment by hmllr** [comment]
this can have different causes .
1. your connection to the OW memory could be bad (this is one of the extension header pins), so make sure your deck is connected with the right orientation and the pins are not bent.
2. you could have something else connected to the OW memory line, like another deck that is broken or the wrong way around or similar
3. the OW memory could be broken

hope this helps,
Hanna

**Reply by meesher1** [reply]
Thanks for the reply. I believe it is issue 3.

I can confirm that the deck is connected/oriented properly and the pins are not bent. I have tested the Flow Deck alone and with the Multi ranger attached, and in both cases I get the error. When I attach just the multi ranger, I get no error. 

Do you believe this is an issue with the hardware of the flow deck itself?

**Reply by hmllr** [reply]
Hmm, hard to say - it could be possible to test it with forcing the flow deck driver. For this you have to run "make menuconfig" in the crazyflie-firmware folder and then enter "bcFlow2" in the "Force load specified custom deck driver" field.
If this works it is only the OW memory.

**Reply by meesher1** [reply]
Thanks for the info. Is there any more documentation to read about this?

**Reply by hmllr** [reply]
There is documentation about how to force a deck: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/howto/ (enabling the driver). Also about the OW memory: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/deck_memory_format/ and there are examples on how to read/write the OW memory, I don't think it should get corrupted normally, but maybe reflashing it is worth a try.

**Reply by meesher1** [reply]
I appreciate all the help so far. I have followed the steps above and still don't think it resolved the issue.
```
CFGBLK: v1, verification [OK]
DECK_INFO: Memory error: wrong header ID
DECK_INFO: Deck 0 has corrupt OW memory. No driver will be initialized!
DECK_INFO: CONFIG_DECK_FORCE=bcFlow2 found
DECK_INFO: compile-time forced driver bcFlow2 added
DECK_CORE: 0 deck(s) found
```

**Reply by hmllr** [reply]
And no other print? I am not completely sure if one broken OW memory kills all drivers, I thought not - but then you should get some init message from the bcFlow2 driver. If that's not the case there is one more thing you can try, that is to disconnect only the OW pin (pin 8 on P2, the right expansion header). If that (while forcing the driver) doesn't work then I am afraid your deck is broken :(
